lib/improver/spotdata/spot_extraction.py

```diff
@@ -202,14 +204,17 @@ def process(self, neighbour_cube, diagnostic_cube, new_title=None):
         spotdata_cube = data_cubes.merge_cube()
 
         # Copy attributes from the diagnostic cube that describe the data's
-        # provenance, and update title if specified
+        # provenance
         spotdata_cube.attributes = diagnostic_cube.attributes
         spotdata_cube.attributes['model_grid_hash'] = (
             neighbour_cube.attributes['model_grid_hash'])
-        if new_title is not None:
-            spotdata_cube.attributes["title"] = new_title
-        else:
-            spotdata_cube.attributes.pop("title", None)
+
+        # Remove grid attributes and update title
+        for attr in MOSG_GRID_ATTRIBUTES:
```

Comment:
This is MO specific and I believe needs to be configurable. That is not entirely trivial if we don't want to clutter interfaces.  We could create a simple module that reads a json and makes the variables accessible in much the same way as has been done here, thus making this configurable on a site-by-site basis, avoiding complicated interfaces, and removing the need to store the MO specific stuff in the code. For example:

```
from improver.utilities.cli_utilities import load_json_or_none
class OrganisationSpecific:
   def __init__(self):
      grid_attributes = load_json_or_none(IMPROVER_ROOT + "/etc/grid_attributes.json")
      if grid_attributes is not None:
         self.GRID_ATTRIBUTES = grid_attributes
      else:
         self.GRID_ATTRIBUTES = {}
```
This could then be imported in this and other modules that need it, e.g OrganisationSpecific.GRID_ATTRIBUTES. The other site specific bits could be in here too, such as MOSG_GRID_DEFINITION.

This is just one proposal as I don't want to throw a non-trivial request at you without some idea. It might be best to talk to someone in the green team about the best approach.

Reply:
I do like the idea of some kind of site init type file for some of these metadata manipulations. I think talking to the green team is a good idea too and I certainly haven't thought through all the implications

Reply:
I think this is follow-on work, and will consult the Green team.  However it is important to note that:

- This has no detrimental effect on non-StaGE files - it does not require attributes, simply ignores if StaGE grid attributes are not there
- All the files we are processing (including processed ECMWF data) has these attributes
- There are no grid-defining attributes mandated by the CF standard, therefore we're not really expecting anything like this except from the special processing StaGE does
- Including this in the code ensures it's covered by automated tests and reduces unnecessary complexity in our suites

So there are a lot worse ways of doing it than this.